### PR TITLE
Fix local-env issue of missing `xfs` user

### PR DESCRIPTION
**Pull request by thelovekesh**
## Summary
As it can be seen in the latest CI runs - https://github.com/ampproject/amp-wp/actions/runs/9389955987/job/25858689875, the `xfs` user is no longer available in the container powered by alpine Linux, and resulting in error like:
```
RROR: for local-env_cli_1  Cannot start service cli: unable to find user xfs: no matching entries in passwd file
```

This PR adds a custom Docker image which extends `wordpress` image and installs the wp-cli binary in the image itself.

```Dockerfile
FROM wordpress

# WP CLI
RUN curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar \
	&& chmod +x wp-cli.phar \
	&& mv wp-cli.phar /usr/local/bin/wp

CMD ["apache2-foreground"]

```

## Checklist

- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: bin/local-env/start.sh [hunk 1]
```diff
@@ -9,12 +9,6 @@ set -e
 # Change to the expected directory
 cd "$(dirname "$0")/../.."
 
-# Check whether Node and NVM are installed
-. "$(dirname "$0")/install-node-nvm.sh"
-
-# Check whether Composer installed
-. "$(dirname "$0")/install-composer.sh"
-
```
Comment (westonruter) [inline bin/local-env/start.sh:17]: No longer needed?

**Review by westonruter — APPROVED** [review]
It's over my head, but LGTM!

File: bin/local-env/start.sh (hunk 1 above)
Comment (thelovekesh) [reply bin/local-env/start.sh:17]: No, the CI runners already have these tools pre-installed, and we configure NodeJS and Composer in the CI. 

For local development, I doubt if this is being used given we already have wp-env setup.

**Comment by thelovekesh** [comment]
E2E tests are running now - https://github.com/ampproject/amp-wp/actions/runs/9391104052/job/25862514993. I'll fix the failing tests in a different PR after merging dependabot PRs.

**Comment by thelovekesh** [comment]
E2E tests are running as expected in local env. QA Passed ✅.

**Pull request merged**